src/raftstore/coprocessor/split_check/keys.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use raftstore::store::{util, Msg};
+use rocksdb::DB;
+use util::transport::{RetryableSendCh, Sender};
+
+use super::super::metrics::*;
+use super::super::{Coprocessor, KeyEntry, ObserverContext, SplitCheckObserver, SplitChecker};
+use super::Host;
+
+pub struct Checker {
+    max_keys: u64,
+    split_keys: u64,
+    current_keys: u64,
+    split_key: Option<Vec<u8>>,
+}
+
+impl Checker {
+    pub fn new(max_keys: u64, split_keys: u64) -> Checker {
+        Checker {
+            max_keys,
+            split_keys,
+            current_keys: 0,
+            split_key: None,
+        }
+    }
+}
+
+impl SplitChecker for Checker {
+    fn on_kv(&mut self, _: &mut ObserverContext, key: &KeyEntry) -> bool {
+        if key.is_commit_version() {
```

Comment:
`is_commit_version` sound like a concept from txn layer.

Reply:
yes, only rows committed by txn can split by keys here. Any other suggestion?

Reply:
Oh, it's okay, because it's in the coprocessor.